File: padroesCriacionais/padrao-de-projeto-prototype/php/src/Npc.php
<?php

namespace Prototype\Php;

class Npc
{
    public function __construct(
        private String $name,
        private String $level,
        private Map $map
    ) {}

    /**
     * @return String
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param String $name
     */
    public function setName(string $name): void
    {
        $this->name = $name;
    }

    /**
     * @return String
     */
    public function getLevel(): string
    {
        return $this->level;
    }

    /**
     * @param String $level
     */
    public function setLevel(string $level): void
    {
        $this->level = $level;
    }

    /**
     * @return Map
     */
    public function getMap(): Map
    {
        return $this->map;
    }

    /**
     * @param Map $map
     */
    public function setMap(Map $map): void
    {
        $this->map = $map;
    }

    public function __clone(): void
    {
        $this->map = clone $this->map;
    }

    public function __toString(): string
    {
        return "Npc{" .
            "name=" . $this->name .
            ", level=" . $this->level .
            ", map='" . $this->map . '\'' .
            '}';
    }
}

File: padroesCriacionais/padrao-de-projeto-prototype/java/src/main/java/Npc.java
public class Npc implements Cloneable {
    private Map map;
    private String name;
    private String level;

    public Npc(String name, String level, Map map) {
        this.name = name;
        this.level = level;
        this.map = map;
    }

    public Map getMap() {
        return map;
    }

    public void setMap(Map map) {
        this.map = map;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    @Override
    public Npc clone() throws CloneNotSupportedException {
        Npc npcClone = (Npc) super.clone();
        npcClone.map = (Map) npcClone.map.clone();
        return npcClone;
    }

    @Override
    public String toString() {
        return "Npc{" +
                "name=" + name +
                ", level=" + level +
                ", map='" + map + '\'' +
                '}';
    }
}
